### ISSUE TITLE
cmake failing

### ISSUE THREAD

**Issue by edouardchalaron** [AD-Vega/qarv#1]
-- Looking for av_pix_fmt_desc_get
-- Looking for av_pix_fmt_desc_get - not found

It seems to belong to libavutils but devel files and libs are on the system.
Or so it seems.
Cheers
E

**Comment by exzombie** [comment]
Hi,

Thanks for reporting this issue. Please provide more information. I looked through the emails you sent and I am still confused. From what I can gather, you have several systems with different versions of Ubuntu and OpenSuse, all with ffmpeg-dev packages installed. First, confirm that the packages you have installed indeed contain the `libavutil/pixdesc.h` file somewhere; having the file in the `libavutil` subdirectory is important! If this is the case, please provide specific versions of operating systems where qarv fails to build. I am currently on a metered internet connection and cannot download entire distributions, but I can still check individual packages to see what is going on.

**Comment by edouardchalaron** [comment]
Hello

It compiles and starts on Ubuntu 15, can't run test it on my laptop
Not compiling on OpenSuse 13.2 and later (not tested on earlier versions)

Pixdesc.h is present on both distribution

on opensuse this file is in 
/usr/include/ffmpeg/libavutil

on Ubuntu
/usr/include/x86_64-linux-gnu/libavutil/

However : on opensuse cmake says it finds libavutil 
but fails here
-- Looking for av_pix_fmt_desc_get - not found

Thanks
E.

**Comment by exzombie** [comment]
Please provide output of

```
pkg-config --cflags libavutil
```

for both distributions.

**Comment by edouardchalaron** [comment]
pkg-config --cflags libavutil

**Opensuse** 13.2 and Leap is
-I/usr/include/ffmpeg 

Cmake

-- checking for module 'libavutil'
--   found libavutil, version 54.31.100
-- Looking for av_pix_fmt_desc_get
-- Looking for av_pix_fmt_desc_get - not found

**Ubuntu** 15 is
-I/usr/include/x86_64-linux-gnu

Thanks
E

**Comment by exzombie** [comment]
Wait, let's take a step back. Are you _sure_ that cmake is failing? Even if `av_pix_fmt_desc_get` is not found, it should still build successfully. I haven't had a system without this function for a while now, let's use the opportunity to make sure qarv can still work without it.

**Comment by edouardchalaron** [comment]
mmmm 
This is where I am lacking enough skills.
I did interpret it as a Cmake failure, because the file is here and cmake is reporting not.

My conclusion was that

Updating '/home/edouard/qarv/build/qarv_sl.qm'...
    Generated 189 translation(s) (189 finished and 0 unfinished)
[  2%] Generating src/filters/moc_levels.cxx
[  3%] Generating src/filters/moc_c41.cxx
[  4%] Generating ui_levels.h
[  5%] Generating ui_c41.h
[  6%] Generating src/recorders/moc_rawrecorders.cxx
[  8%] Generating src/recorders/moc_gstrecorders.cxx
[  9%] Generating src/recorders/moc_imagerecorder.cxx
[ 10%] Generating src/decoders/moc_mono12packed.cxx
[ 11%] Generating src/decoders/moc_monounpackeddecoders.cxx
[ 12%] Generating src/decoders/moc_bayer.cxx
[ 13%] Generating src/moc_glvideowidget.cxx
[ 15%] Generating src/moc_glhistogramwidget.cxx
[ 16%] Generating src/moc_qarvmainwindow.cxx
[ 17%] Generating src/moc_roicombobox.cxx
[ 18%] Generating src/moc_globals.cxx
[ 19%] Generating src/moc_workthread.cxx
[ 20%] Generating src/api/moc_qarvtype.cxx
[ 22%] Generating src/api/moc_qarvgui.cxx
[ 23%] Generating src/api/moc_qarvcamera.cxx
[ 24%] Generating src/api/moc_qarvcameradelegate.cxx
[ 25%] Generating src/filters/moc_filter.cxx
[ 26%] Generating ui_mainwindow.h
[ 27%] Generating qrc_icons.cxx
Scanning dependencies of target qarv-2
[ 29%] Building CXX object CMakeFiles/qarv-2.dir/src/filters/moc_levels.cxx.o
[ 30%] Building CXX object CMakeFiles/qarv-2.dir/src/filters/moc_c41.cxx.o
[ 31%] Building CXX object CMakeFiles/qarv-2.dir/src/filters/levels.cpp.o
[ 32%] Building CXX object CMakeFiles/qarv-2.dir/src/filters/c41.cpp.o
[ 33%] Building CXX object CMakeFiles/qarv-2.dir/src/recorders/moc_rawrecorders.cxx.o
In file included from /home/edouard/qarv/build/src/recorders/../../../src/recorders/recorder.h:23:0,
                 from /home/edouard/qarv/build/src/recorders/../../../src/recorders/rawrecorders.h:23,
                 from /home/edouard/qarv/build/src/recorders/moc_rawrecorders.cxx:9:
/home/edouard/qarv/src/api/qarvdecoder.h:30:30: fatal error: libavutil/pixfmt.h: No such file or directory
 #include <libavutil/pixfmt.h>
                              ^
compilation terminated.
CMakeFiles/qarv-2.dir/build.make:276: recipe for target 'CMakeFiles/qarv-2.dir/src/recorders/moc_rawrecorders.cxx.o' failed
make[2]: **\* [CMakeFiles/qarv-2.dir/src/recorders/moc_rawrecorders.cxx.o] Error 1
CMakeFiles/Makefile2:99: recipe for target 'CMakeFiles/qarv-2.dir/all' failed
make[1]: **\* [CMakeFiles/qarv-2.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: **\* [all] Error 2

was the consequence of cmake missing the file.
From your mail I suspect I might be in the wrong ...

**Comment by exzombie** [comment]
Ok, this makes more sense. I pushed a change to your c41 branch, please test.

**Comment by edouardchalaron** [comment]
Hello 
Thanks for that. I fighting with git ... forgot the commands
I did 
git clone for master
git checkout -b edouard
git fetch

but c41 does not appear ... sorry for being so dumb with this

**Comment by exzombie** [comment]
From where you are now, do

```
git checkout master
git branch -D edouard
git checkout -b edouard origin/edouard
```

An do use a clean build directory ;)

**Comment by edouardchalaron** [comment]
Ok done .. all good !!!!
done remotely with SSH so will test run tomorrow
I will report ;-)
So many thanks
E

**Comment by exzombie** [comment]
Great! Fixed in 1241f235609ce835f5024a3c1e1c8b2e3b204da7 on master. Note that I have renamed the commit in your branch, which means that you will have to first delete it, do a fetch and then checkout again.

Thanks for your help!

**Comment by edouardchalaron** [comment]
Thanks to you too !!
